app/gui/src/presenter/searcher.rs

```diff
@@ -361,6 +361,13 @@ impl Searcher {
 
                     entry_selected <- grid.active.filter_map(|&s| s?.as_entry_id());
                     entry_hovered <- grid.hovered.map(|&s| s?.as_entry_id());
+                    hovered_not_selected <- all_with(&grid.hovered, &grid.active, |h, s| {
+                        match (h, s) {
+                            (Some(h), Some(s)) => h.as_entry_id() != s.as_entry_id(),
+                            _ => false,
+                        }
+                    });
```

Comment:
Alternatively, it could be something like `|h.map(|h| h.as_entry_id()) == s.map(|s| s.as_entry_id())` but I'm not sure what's better.

Reply:
Honestly, I'd prefer `match`, as I can't understand what happens in your example.